File: algorithm/python/backjoon/implement/roomnumber.py
import sys
from math import ceil

In = sys.stdin.readline
room = In().strip() # 9999
cnt = [0]*9

for i in room:
    num = int(i)
    if num == 6 or num == 9:
        cnt[6] += 0.5
    else:
        cnt[num] += 1

cnt[6] = ceil(cnt[6])
print(int(max(cnt)))

File: algorithm/java/backjoon/implement/roomnumber.java
package backjoon.implement;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

class roomnumber {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String room = br.readLine();
        double[] cnt = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0};

        for (int i=0; i < room.length(); i++) {
            int num = room.charAt(i) - '0';
            if (num == 6 || num == 9) {
                cnt[6] += 0.5;
            } else {
                cnt[num] += 1;
            }
        }
        cnt[6] = (int) Math.ceil(cnt[6]);
        Arrays.sort(cnt);
        int answer = (int) cnt[cnt.length-1];
        System.out.println(answer);

    }
}
